### PR TITLE
Add code to support creating storage with separate reader/writer

### PR DESCRIPTION
**Pull request by YuchenWang01**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.
_____________________________________________________________________________________
This PR adds a ProvideRW method which would return a storage with separate reader/writer. The reader field can be empty, in which case it would only connects to the writer.
The user needs to implement the CreateRW method for StorageCreator interface based on the backend storage types (e.g. PostgreSQL, MySQL, etc.).

### CODE REVIEW

File: go/v1beta1/storage/storage.go [hunk 1]
```diff
@@ -95,6 +96,37 @@ func (p GrafeasStorageProvider) Provide(_ string, confi *config.StorageConfigura
...
+// ProvideRW returns a storage which is configured based on the receiver's fields. The storage connects to different reader/writer. If no reader is provided, then it will only connect to the writer.
+func (p GrafeasStorageProvider) ProvideRW(_ string, confi *config.StorageConfiguration) (*storage.Storage, error) {
+	conf, err := rdsconfig.New(confi)
+	if err != nil {
+		return nil, fmt.Errorf("%s, err: %v", errMsgInitConfig, err)
+	}
+
+	// TODO: Use the context passed from main after
+	// the signature of RegisterStorageTypeProvider is updated to include it.
+	writerConnector, err := newConnector(context.Background(), conf, p.drv, p.credentialsCreator, log.Default(), "")
```
Comment (alan4chen) [inline go/v1beta1/storage/storage.go:108]: Do we need to pass `conf.Host` to `newConnector` for writerConnector?

**Review by alan4chen — APPROVED** [review]

File: go/v1beta1/storage/storage.go (hunk 1 above)
Comment (YuchenWang01) [reply go/v1beta1/storage/storage.go:108]: conf.Host is the default value. The parameter is used to overwrite the host.

**Review by hkadakia — APPROVED** [review]
Thanks. lgtm 👍🏼

File: go/v1beta1/storage/connector.go [hunk 2]
```diff
@@ -49,9 +49,13 @@ type connector struct {
 	dsnLock sync.RWMutex
 }
 
-func newConnector(ctx context.Context, conf *config.Config, driver driver.Driver, cc CredentialsCreator, logger *log.Logger) (*connector, error) {
+func newConnector(ctx context.Context, conf *config.Config, driver driver.Driver, cc CredentialsCreator, logger *log.Logger, overwriteHost string) (*connector, error) {
```
Comment (zhaoyonghe) [inline go/v1beta1/storage/connector.go:52]: I do not like the parameter `overwiteHost`. It is better (though not the best) to have a boolean parameter to switch the reader/writer instance.

**Pull request merged**